### PR TITLE
Add unit tests for hint utils

### PR DESCRIPTION
**Pull request by tcoratger**
## Add unit tests for hint utils

This pull request adds unit tests for the hint utils module to ensure its functionality is thoroughly tested.

### Changes Made
- Added unit tests covering various functions in the hint utils module.
- Verified the behavior of each function with valid and invalid inputs.
- Ensured appropriate error handling and edge case coverage.

### CODE REVIEW

File: src/hint_processor/hint_utils.zig [hunk 1]
```diff
@@ -341,3 +367,94 @@ test "getPtrFromVarName: invalid" {
...
+    try ids_data.put("value", HintReference.initSimple(-8));
+
+    // Invokes `getPtrFromVarName` to retrieve the pointer from the "value" variable, expecting an error.
+    try expectEqual(
+        HintError.UnknownIdentifier,
+        getRelocatableFromVarName("value", &vm, ids_data, .{}),
+    );
+}
+
+test "getConstantFromVarName: valid without point in the middle of var name" {
```
Comment (lana-shanghai) [inline src/hint_processor/hint_utils.zig:423]: isn't it the other way around? with / without the period?

**Review by lana-shanghai — APPROVED** [review]
lgtm

File: src/hint_processor/hint_utils.zig (hunk 1 above)
Comment (tcoratger) [reply src/hint_processor/hint_utils.zig:423]: Don't think so because we have to handle the point case which means that for `ids.MAX_LOW` we must capture `MAX_LOW` (which is tested in the following unit test). In this specific test, I am testing the simplest case (without any point in the middle) with only `value` which must be captured correctly. I also added `ids.MAX_LOW` in the hashmap but it is of no use, just to check that it does not pollute the test.
